src/main/resources/META-INF/m2e/lifecycle-mapping-metadata.xml

```diff
@@ -0,0 +1,68 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<lifecycleMappingMetadata>
```

Comment:
Specific only for Eclipse ...

Reply:
could be used by any IDE supporting incremental builds, but currently m2e is the only consumer, right.

Reply:
Eclipse and its derivative, such as Java edition in VSCode.

Reply:
is there option to not store such specific IDE file here?
I don't like much this... I would be -1.
we already have a sort of specific API and now specific files.
Imagine if every IDE ask for storing their specific files?

Reply:
Fact is: There is no other metadata for any other IDE. Every IDE could reuse this metadata. We already have m2e metadata e.g. in https://github.com/apache/maven-resources-plugin/blob/master/src/main/resources/META-INF/m2e/lifecycle-mapping-metadata.xml.

Reply:
which should have never been accepted....

Reply:
> Imagine if every IDE ask for storing their specific files? sounds a bad design to me.

Indeed that would be bad and was the reason why I/we the M2E team reached out to the Maven devs on their mailing list (the link @kwin posted above) to establish a unified API defined/maintained by Maven that all IDEs could use. Eventually this probably would have lead to a similar Maven-defined/maintained lifecycle-mapping metadata format (which actually also could be generated from annotations see e.g. https://github.com/eclipse-m2e/m2e-core/issues/830).
But there was only negative feedback from all Maven devs about that suggestion, so each IDE has to continue to handle it their one way.

Anyway, as @mickaelistria said below we can maintain that metadata in a Eclipse M2E plug-in.

Reply:
> Indeed that would be bad and was the reason why I/we the M2E team reached out to the Maven devs on their mailing list (the link @kwin posted above) to establish a unified API defined/maintained by Maven that all IDEs could use. [...] But there was only negative feedback from all Maven devs about that suggestion, so each IDE has to continue to handle it their one way.

A very very strange conclusion drawn from a ML thread having his post: https://lists.apache.org/thread/044tno4lvdjvm4yv7orsqpmn3x4175c4

Reply:
That's right Guillaume Nodet was not averse to the suggestion, but from the other not so positive comments I had the impression that this was not the general conclusion. However, if that was a misinterpretation from my side and the suggestion will be considered in the future, I'm glad. :)

Reply:
As there was no consensus to add m2e specific metadata I removed it now in https://github.com/apache/maven-dependency-plugin/pull/257/commits/c34c3c754433d2a6805fd98eb60d6874222ed3dd.